pkg/fanal/analyzer/pkg/dpkg/copyright.go

```diff
@@ -0,0 +1,131 @@
+package dpkg
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	classifier "github.com/google/licenseclassifier/v2"
+	"github.com/google/licenseclassifier/v2/assets"
+	"github.com/samber/lo"
+	"golang.org/x/exp/slices"
+	"golang.org/x/xerrors"
+
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+)
+
+func init() {
+	analyzer.RegisterAnalyzer(&dpkgLicenseAnalyzer{})
+
+	var err error
+	licenseClassifier, err = assets.DefaultClassifier()
+	if err != nil {
+		panic(err)
+	}
+}
+
+var (
+	dpkgLicenseAnalyzerVersion = 1
+
+	licenseClassifier *classifier.Classifier
+
+	commonLicenseReferenceRegexp = regexp.MustCompile(`/?usr/share/common-licenses/([0-9A-Za-z_.+-]+[0-9A-Za-z+])`)
+)
+
+// dpkgLicenseAnalyzer parses copyright files and detect licenses
+type dpkgLicenseAnalyzer struct{}
+
+// Analyze parses /usr/share/doc/*/copyright files
+func (a dpkgLicenseAnalyzer) Analyze(_ context.Context, input analyzer.AnalysisInput) (*analyzer.AnalysisResult, error) {
+	licenses, err := a.parseCopyright(input.Content)
+	if err != nil {
+		return nil, xerrors.Errorf("parse copyright %s: %w", input.FilePath, err)
+	} else if len(licenses) == 0 {
+		return nil, nil
+	}
+
+	findings := lo.Map(licenses, func(license string, _ int) types.LicenseFinding {
+		return types.LicenseFinding{License: license}
+	})
+
+	// e.g. "usr/share/doc/zlib1g/copyright" => "zlib1g"
+	pkgName := strings.Split(input.FilePath, "/")[3]
+
+	return &analyzer.AnalysisResult{
+		Licenses: []types.LicenseFile{
+			{
+				Type:     types.LicenseTypeDpkg,
+				FilePath: input.FilePath,
+				Findings: findings,
+				Package:  pkgName,
+			},
+		},
+	}, nil
+}
+
+// parseCopyright parses /usr/share/doc/*/copyright files
+func (a dpkgLicenseAnalyzer) parseCopyright(r dio.ReadSeekerAt) ([]string, error) {
+	scanner := bufio.NewScanner(r)
+	var licenses []string
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		switch {
+		case strings.HasPrefix(line, "License:"):
+			// Machine-readable format
+			// cf. https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/#:~:text=The%20debian%2Fcopyright%20file%20must,in%20the%20Debian%20Policy%20Manual.
+			l := strings.TrimSpace(line[8:])
+			if !slices.Contains(licenses, l) {
```

Comment:
```suggestion
			if len(l) > 0 && !slices.Contains(licenses, l) {
```

Some packages don't have short license name and only contain license text in remaining lines. 
e.g. `liblzma5/copyright` : 
```
...
License:
 Different licenses apply to different files in this package. Here
 is a rough summary of which licenses apply to which parts of this
 package (but check the individual files to be sure!):
...
```
In this case we should check this license with `licenseclassifier`.